flink-ml-lib/src/main/java/org/apache/flink/ml/feature/stringindexer/StringIndexerModel.java

```diff
@@ -0,0 +1,198 @@
+                                            broadcastModelKey, inputCols, getHandleInvalid()),
+                                    outputTypeInfo);
+                        });
+        return new Table[] {tEnv.fromDataStream(result)};
+    }
+
+    /** Maps the input columns to integer values according to the model data. */
+    private static class String2Index extends RichFlatMapFunction<Row, Row> {
+        private HashMap<String, Integer>[] modelDataMap;
+        private final String broadcastModelKey;
+        private final String[] inputCols;
+        private final String handleInValid;
+
+        public String2Index(String broadcastModelKey, String[] inputCols, String handleInValid) {
+            this.broadcastModelKey = broadcastModelKey;
+            this.inputCols = inputCols;
+            this.handleInValid = handleInValid;
+        }
+
+        @Override
+        @SuppressWarnings("unchecked")
+        public void flatMap(Row input, Collector<Row> out) {
+            if (modelDataMap == null) {
+                modelDataMap = new HashMap[inputCols.length];
+                StringIndexerModelData modelData =
+                        (StringIndexerModelData)
+                                getRuntimeContext().getBroadcastVariable(broadcastModelKey).get(0);
+                String[][] stringsArray = modelData.stringArrays;
+                for (int i = 0; i < stringsArray.length; i++) {
+                    int idx = 0;
+                    modelDataMap[i] = new HashMap<>(stringsArray[i].length);
+                    for (String string : stringsArray[i]) {
+                        modelDataMap[i].put(string, idx++);
+                    }
+                }
+            }
+            Row outputIndices = new Row(inputCols.length);
+            for (int i = 0; i < inputCols.length; i++) {
+                Object objVal = input.getField(inputCols[i]);
+                String stringVal;
+                if (objVal instanceof String) {
+                    stringVal = (String) objVal;
+                } else if (objVal instanceof Number) {
+                    stringVal = String.valueOf(objVal);
+                } else {
+                    throw new RuntimeException(
+                            "The input column only supports string and numeric type.");
+                }
+                if (modelDataMap[i].containsKey(stringVal)) {
+                    outputIndices.setField(i, modelDataMap[i].get(stringVal));
+                } else {
+                    switch (handleInValid) {
+                        case StringIndexerModelParams.SKIP_INVALID:
+                            return;
+                        case StringIndexerModelParams.ERROR_INVALID:
+                            throw new RuntimeException(
+                                    "The input contains unseen string: "
+                                            + stringVal
+                                            + ". To handle unseen strings, set Param "
```

Comment:
The error message seems incorrect. In order to `handle` the unseen strings, any one of three options work, right?

I guess you mean that `To keep unseen strings in the output, set ...`. But if we say so, do we also need to say `To drop unseen strings from the output, set ...`? Would that be too verbose?

It is probably simpler to just mention "see handleInvalid parameter for more options".

Reply:
Sounds good :)